proof/infoflow/refine/RISCV64/ArchADT_IF_Refine.thy

```diff
@@ -197,25 +197,25 @@ lemma doUserOp_if_invs'[ADT_IF_Refine_assms, wp]:
 lemma doUserOp_valid_duplicates[ADT_IF_Refine_assms, wp]:
   "doUserOp_if f tc \<lbrace>arch_extras\<rbrace>"
   apply (simp add: doUserOp_if_def split_def)
-  apply (wp dmo_invs' select_wp | simp)+
+  apply (wp dmo_invs' | simp)+
   done
 
 lemma doUserOp_if_schedact[ADT_IF_Refine_assms, wp]:
   "doUserOp_if f tc \<lbrace>\<lambda>s. P (ksSchedulerAction s)\<rbrace>"
   apply (simp add: doUserOp_if_def)
-  apply (wp select_wp | wpc | simp)+
+  apply (wp | wpc | simp)+
   done
 
 lemma doUserOp_if_st_tcb_at[ADT_IF_Refine_assms, wp]:
    "doUserOp_if f tc \<lbrace>st_tcb_at' st t\<rbrace>"
   apply (simp add: doUserOp_if_def)
-  apply (wp select_wp | wpc | simp)+
+  apply (wp | wpc | simp)+
   done
 
 lemma doUserOp_if_cur_thread[ADT_IF_Refine_assms, wp]:
   "doUserOp_if f tc \<lbrace>\<lambda>s. P (ksCurThread s)\<rbrace>"
   apply (simp add: doUserOp_if_def)
-  apply (wp select_wp | wpc | simp)+
+  apply (wp | wpc | simp)+
```

Comment:
probably too late now, but isn't this equivalent to `wpsimp+` or `wpsimp`?

Reply:
Yep, it definitely is. In cases like this where it really was just deleting the rule from wp, the process I was using to make these changes made it easier to just keep on moving. Beyond that, it was basically random which lemmas I would stop at and clean up and which I would just remove the rules and move on.